src/compute/sort/row/interner.rs

```diff
@@ -0,0 +1,431 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::hash::{BuildHasher, Hash};
+use std::num::NonZeroU32;
+use std::ops::Index;
+
+use hashbrown::hash_map::RawEntryMut;
+use hashbrown::HashMap;
```

Comment:
To my best knowledge, rust std's `HashMap` uses the same implementation as `hashbrown`. Is there any reason to hashbrown directly?

Reply:
> Is there any reason to hashbrown directly?

No, it just follows arrow-rs.

Reply:
How about using std's `HashMap` instead? So we don't need to introduce `hashbrown`.

Reply:
At least historically the raw entry API, which this requires, isn't exposed in std

Reply:
> At least historically the raw entry API, which this requires to store the values separately from the HashMap, isn't exposed in std

Thanks for explanation!

Is the raw entry API we use here something different from https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html?

Reply:
By the way, we have https://doc.rust-lang.org/std/collections/hash_map/enum.RawEntryMut.html under nightly feature `hash_raw_entry`.